lib/core-integration/src/Test/Integration/Scenario/API/Shelley/StakePools.hs

```diff
@@ -823,7 +820,7 @@ spec = describe "SHELLEY_STAKE_POOLS" $ do
     describe "STAKE_POOLS_JOIN_01x - Fee boundary values" $ do
         it "STAKE_POOLS_JOIN_01x - \
             \I can join if I have just the right amount" $ \ctx -> runResourceT $ do
-            w <- fixtureWalletWith @n ctx [costOfJoining ctx + depositAmt ctx]
+            w <- fixtureWalletWith @n ctx [costOfJoining ctx + depositAmt ctx + minUTxOValue]
```

Comment:
Note: a little downside of the new unified approach for coin selection is that the algorithm must always be able to create a change output. Which means that, even is the wallet contains _just_ enough to pay for fees and deposit, one would still get an error if there's not enough to cover for the min utxo value of the change output. The actual cost of the transaction is unchanged however, because the change goes back to the wallet.